objc/objc_runtime_darwin.go

```diff
@@ -215,19 +215,18 @@ func (i Ivar) Offset() uintptr {
 	return ret
 }
 
-// _IMP is unexported so that the only way to make this type is by providing a Go function and casting
-// it with the IMP function
-type _IMP uintptr
+// IMP is a function pointer that can be called by Objective-C code.
+type IMP uintptr
 
-// IMP takes a Go function that takes (ID, SEL) as its first two arguments. It returns an _IMP function
+// NewIMP takes a Go function that takes (ID, SEL) as its first two arguments. It returns an IMP function
 // pointer that can be called by Objective-C code. The function pointer is never deallocated.
-func IMP(fn interface{}) _IMP {
+func NewIMP(fn interface{}) IMP {
 	// this is only here so that it is easier to port C code to Go.
 	// this is not guaranteed to be here forever so make sure to port your callbacks to Go
 	// If you have a C function pointer cast it to a uintptr before passing it
 	// to this function.
```

Comment:
Do we want to make this part of the exported API? Or just leave it as internal

Reply:
I think we should make it exported, or how do users use this?

Reply:
Or do we not care since you can now cast a uintptr to IMP?

Reply:
It would be used like `NewIMP(uintptr(C.someCFunction))`

Reply:
Let's go back to discuss the idea https://github.com/ebitengine/purego/issues/22

Reply:
Updated: I have removed uintptr usage.